pkg/new_storage/s3.go

```diff
@@ -59,15 +61,29 @@ func (s *S3) Connect() error {
 
 	awsDefaults := defaults.Get()
 	defaultCredProviders := defaults.CredProviders(awsDefaults.Config, awsDefaults.Handlers)
+	customCredProviders := defaultCredProviders
 
-	// Define custom static cred provider
-	staticCreds := &credentials.StaticProvider{Value: credentials.Value{
-		AccessKeyID:     s.Config.AccessKey,
-		SecretAccessKey: s.Config.SecretKey,
-	}}
+	if s.Config.AccessKey != "" && s.Config.SecretKey != "" {
+		// Define custom static cred provider
+		staticCreds := &credentials.StaticProvider{Value: credentials.Value{
+			AccessKeyID:     s.Config.AccessKey,
+			SecretAccessKey: s.Config.SecretKey,
+		}}
+		// Append static creds to the defaults
+		customCredProviders = append([]credentials.Provider{staticCreds}, customCredProviders...)
+	}
+
+	awsRoleARN := os.Getenv("AWS_ROLE_ARN")
+	awsWebIdentityTokenFile := os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE")
+	if awsRoleARN != "" && awsWebIdentityTokenFile != "" {
+		cfg := &aws.Config{
+			Region: aws.String(s.Config.Region),
+		}
+		stsSvc := sts.New(session.Must(session.NewSession(cfg)))
+		stsProvider := stscreds.NewWebIdentityRoleProvider(stsSvc, awsRoleARN, "", awsWebIdentityTokenFile)
```

Comment:
what is "" magic contant here?

Reply:
> What is `""` magic constant here?

Session name. When it's an empty string, the AWS SDK will use a timestamp based session name to generate a unique one.